presto-main/src/test/java/io/prestosql/operator/TestDynamicFilterSourceOperator.java

```diff
@@ -255,6 +266,77 @@ public void testCollectWithRealNaN()
                         new DynamicFilterId("0"), Domain.multipleValues(REAL, ImmutableList.of((long) floatToRawIntBits(42.0f)))))));
     }
 
+    @Test
+    public void testCollectTooMuchRowsDouble()
+    {
+        int maxRowCount = getDynamicFilteringMaxPerDriverRowCount(pipelineContext.getSession());
+        OperatorFactory operatorFactory = createOperatorFactory(channel(0, DOUBLE));
+        verifyPassthrough(createOperator(operatorFactory),
+                ImmutableList.of(DOUBLE),
+                new Page(createDoubleSequenceBlock(0, maxRowCount + 1)),
+                new Page(createDoubleRepeatBlock(Double.NaN, maxRowCount + 1)));
+        operatorFactory.noMoreOperators();
+
+        assertEquals(partitions.build(), ImmutableList.of(TupleDomain.all()));
+    }
+
+    @Test
+    public void testCollectTooMuchRowsReal()
+    {
+        int maxRowCount = getDynamicFilteringMaxPerDriverRowCount(pipelineContext.getSession());
+        OperatorFactory operatorFactory = createOperatorFactory(channel(0, REAL));
+        verifyPassthrough(createOperator(operatorFactory),
+                ImmutableList.of(REAL),
+                new Page(createSequenceBlockOfReal(0, maxRowCount + 1)),
+                new Page(createBlockOfReals(Collections.nCopies(maxRowCount + 1, Float.NaN))));
+        operatorFactory.noMoreOperators();
+
+        assertEquals(partitions.build(), ImmutableList.of(TupleDomain.all()));
+    }
+
+    @Test
+    public void testCollectTooMuchRowsNonOrderable()
```

Comment:
please also add a test where there are too much rows, but collection of min/max is disabled

Reply:
Added `testCollectRowsNonOrderableWithoutMinMax`